src/charm.py

```diff
@@ -142,12 +142,27 @@ def __init__(self, *args):
         self._observer.start_observer()
         self._grafana_agent = COSAgentProvider(
             self,
-            metrics_endpoints=[
-                {"path": "/metrics", "port": METRICS_PORT},
+            metrics_endpoints=[{"path": "/metrics", "port": METRICS_PORT}],
+            scrape_configs=self.patroni_scrape_config,
+            refresh_events=[
+                self.on[PEER].relation_changed,
+                self.on.secret_changed,
+                self.on.secret_remove,
```

Comment:
`certificates` rel events don't guarantee that the scheme will switch, so we update on events that can potentially change or remove the certs